src/rules.ts

```diff
@@ -1658,6 +1659,168 @@ export const rules: Rule[] = [
       [new TextOption('Title Key', 'Which YAML key to use for title', 'title')],
   ),
 
+  new Rule(
+      'YAML Title Alias',
+      'Inserts the title of the file into the YAML frontmatter\'s aliases section. Gets the title from the first H1 or filename.',
+      RuleType.YAML,
+      (text: string, options = {}) => {
+        text = initYAML(text);
+        let title = ignoreCodeBlocksYAMLTagsAndLinks(text, (text) => {
+          const result = text.match(/^#\s+(.*)/m);
+          if (result) {
+            return result[1];
+          }
+          return '';
+        });
+        title = title || options['metadata: file name'];
+
+        let yaml = text.match(yamlRegex)[1];
+        const aliasesRegex = /(?<=^|\n)aliases:[ \t]*((?:\S|\n)(?:.|\n)+?)\n(?=$|\S)/;
+        let aliasesMatch = yaml.match(aliasesRegex);
+
+        const linterMarkerComment = '# linter-yaml-title-alias';
+
+        if (!aliasesMatch) {
+          let emptyValue;
+          switch (options['YAML aliases new section style']) {
+            case 'Multi-line array':
+              emptyValue = '\n  - \'\'';
+              break;
+            case 'Single-line array':
+              emptyValue = ' [\'\']';
+              break;
+            case 'Single string':
+              emptyValue = ' \'\'';
+              break;
+            default:
+              throw new Error(`Unsupported setting 'YAML aliases new section style': ${options['YAML aliases new section style']}`);
+          }
+
+          const newAliasesSection = `aliases:${emptyValue} ${linterMarkerComment}\n`;
```

Comment:
How does this work properly for `Single string` and `Single-line array`?

It looks like it will just add it at the end of the array which will need to be enforced and thus the comment is unnecessary. If it is the single line array it will likely add it at the end of the entry, but what keeps other values from being added after the comment? 

I guess all of that leads me to ask if it makes sense to not just enforce the order of where the alias will be in the list instead of using a comment since it seems to be ineffective for single-line arrays and single line strings?

Reply:
To expound on the above, if we are saying the first element in the array or the list of strings is the value is there not a better way to check if the value existed prior?

Reply:
This works properly, based on my unit tests. This piece of code is needed to create dummy aliases section if it doesn't exist or empty. It is being populated with proper values below. It is made like that to follow DRY principle

Reply:
@pjkaufman I don't understand how are you going to check if the value existed prior? In the `modify` event of the note, we don't have an access to the previous version of the note, that's why I came up with this custom comment idea

Reply:
I am trying to wrap my head around it as well. I am not certain there is a better option. However it seems like it is just an indicator that the value existed prior.

Reply:
However, I did not see a test that checked what would happen if nothing had changed. That is a test case I am a little concerned about given the code I pointed out.

Reply:
Yes, you are totally right, as it just an indicator, I will rewrite it with the custom yaml property. It is way more reliable than the custom comment